plugins/catalog-backend-module-github/src/providers/GithubOrgEntityProvider.ts

```diff
@@ -224,6 +243,202 @@ export class GithubOrgEntityProvider implements EntityProvider {
+        combinedSlug: `${org}/${slug}`,
+        description: description || undefined,
+        parentTeam: { slug: event.team?.parent?.slug || '' } as GithubTeam,
+        // entity will be removed
+        members: [],
+      },
+      {
+        org,
+        client,
+        query: '',
+      },
+    )) as Entity;
+
+    const { added, removed } = createDeltaOperation(org, [group]);
+
+    await this.connection.applyMutation({
+      type: 'delta',
+      removed,
+      added,
+    });
+  }
+
+  private async onTeamEditedInOrganization(event: TeamEvent) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const { slug } = event.team;
+    const org = event.organization.login;
+
+    await this.connection.refresh({
+      keys: [
+        `url:https://${this.options.gitHubConfig.host}/orgs/${org}/teams/${slug}`,
+      ],
+    });
+  }
+
+  private async onMemberChangeInOrganization(
+    event: OrganizationMemberAddedEvent | OrganizationMemberRemovedEvent,
+    createDeltaOperation: DeltaOperationFactory,
+  ) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const userTransformer =
+      this.options.userTransformer || defaultUserTransformer;
+    const { name, avatar_url: avatarUrl, email, login } = event.membership.user;
+    const org = event.organization.login;
+    const { headers } = await this.credentialsProvider.getCredentials({
+      url: this.options.orgUrl,
+    });
+    const client = graphql.defaults({
+      baseUrl: this.options.gitHubConfig.apiBaseUrl,
+      headers,
+    });
+
+    const user = (await userTransformer(
+      {
+        name,
+        avatarUrl,
+        login,
+        email: email || undefined,
+        // we don't have this information in the event, so the refresh will handle that for us
+        organizationVerifiedDomainEmails: [],
+      },
+      {
+        org,
+        client,
+        query: '',
+      },
+    )) as Entity;
+
+    const { added, removed } = createDeltaOperation(org, [user]);
+    await this.connection.applyMutation({
+      type: 'delta',
+      removed,
+      added,
+    });
+  }
+
+  private async onMemberChangeToTeam(event: MembershipEvent) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    // The docs are saying I will receive the slug for the removed event, but the types don't reflect that,
+    // so I will just check to be sure the slug is there
+    // https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#membership
+    if (!('slug' in event.team)) {
+      return;
+    }
+    const { slug: teamSlug } = event.team;
+    const { login } = event.member;
+    const org = event.organization.login;
+
+    await this.connection.refresh({
+      keys: [
+        `url:https://${this.options.gitHubConfig.host}/orgs/${org}/teams/${teamSlug}`,
+        `url:https://${this.options.gitHubConfig.host}/${login}`,
```

Comment:
I put this refresh here, but I don't know why the group and the user stay without the refresh. Is the key wrong? 

@kissmikijr could you help me here please? I am not seeing what I am doing wrong

Reply:
Unfortunately it won't work for the current implementation of the org entity provider. The connection.refresh() works when there is a matching refresh_key for the entity. Currently this refresh_key is added by the UrlReaderProcessor to the Location entities. In the case of this provider it doesn't use Locations for the individual teams/users. The way it works is that it directly provides the raw entities, not the Locations for them.

I think here we will need to rely on the catalogApi and handle the event by either updating an already existing group/user or creating new ones when the event comes in.

Reply:
Thanks, I will change to use catalogApi in this case